rfcs/0001-masonry-backend.md

```diff
@@ -0,0 +1,166 @@
+        Button::new("reset", |data| *data = 0),
+    ))
+}
+```
+
+In that example, `app_logic()` returns a `Column` view which contains two `Button` views. These store the minimum information to represent the concept of two buttons stored in a column.
+
+The `View` trait has an `Element` associated type, which depends on the platform layer, and represents a persisten UI entity. Every time app data changes, a new View is produced. That View is diffed against the app's previous View, and all changes are applied to matching elements: if the new view has a child view that wasn't in the previous version, a new element is added; if the new view no longer has a child view that was in the previous version, the matching element is removed, and so on.
+
+(In web frameworks that use a virtual DOM, this process is sometimes known as "reconciliation".)
+
+Xilem has multiple platform layers. `xilem_web` is based on the web DOM, whereas `xilem_native` is based on the Masonry crate, which provides a native widget tree and some Developer Experience features.
+
+Note that, while the platform layer is sometimes referred to as a "backend", it's somewhat tightly coupled with the reactive layer.
+
+`xilem_web`, `xilem_native` and any other platforms will each have their own version of the `View` trait, the `ViewSequence` trait, the `AnyView` trait, the `BoxedView` type, and so on.
+
+### `xilem_native`
+
+The element type in `xilem_native` must implement `masonry::Widget`.
+
+Widget in Masonry is a trait that represent units of graphical composition, things like text and buttons and boxes and containers. To implement Widget, a type must:
+
+- Be `'static`.
+- Provide info about its children.
+- Handle user events.
+- Compute its layout.
+- Draw itself in a paint target.
+
+See the [Widget trait definition](https://github.com/PoignardAzur/masonry-rs/blob/3cfc2cddec05d48a31a9812bdda27792d336f03f/src/widget/widget.rs#L65-L199) for more details.
+
+
+### Directory structure
+
+The `https://github.com/linebender/xilem` repository has the following crates in its `crates/` directory:
+
+- `xilem_core/`
+- `xilem_web/`
+- `xilem_native/`
+- `masonry/`
+- `xilem/`, which would only be a placeholder crate pointing to `xilem_web/` or `xilem_native/`.
+
+
+## Implementation strategy
+
+The implementation will require several steps:
+
+- Refactoring Masonry to use the same dependencies Xilem currently uses, especially Glazier, Vello, and Parley insted of druid-shell and Piet.
+- Integrating AccessKit support.
+- Moving the Masonry codebase to the Xilem repository, ideally in a way that preserves commit history.
+- Porting the list of Github Issues in the masonry-rs repository to the Xilem repository.
+- Creating a new `xilem_native` crate on parity with the current Xilem crate.
+- Removing the current xilem crate.
+- Publishing both `xilem_native` and the placeholder `xilem` crate on crates.io.
+- Sharing ownership of the crates with Raph Levien.
```

Comment:
I'd like to avoid over-specialising on the release process, see e.g. [#linebender > Our Ideal Crate Release Process](https://xi.zulipchat.com/#narrow/stream/419691-linebender/topic/Our.20Ideal.20crate.20release.20processs).

Reply:
Agree, the RFC feels like it should be multiple RFCs, as it's not just about masonry as a backend.

Reply:
Right. I'll move that part to "future possibilities".